File: PySpark_Crypto/javaSol/schemaClasses/PriceData.java
package schemaClasses;

import java.io.Serializable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class PriceData implements Serializable {

	private static final long serialVersionUID = 1L;

	@SerializedName("close")
	@Expose
	private double close;
	@SerializedName("high")
	@Expose
	private double high;
	@SerializedName("low")
	@Expose
	private double low;
	@SerializedName("open")
	@Expose
	private double open;
	@SerializedName("volume")
	@Expose
	private double volume;

	private double profit;
	
	public double getOpen() {
		return open;
	}

	public void setOpen(double open) {
		this.open = open;
	}

	public double getHigh() {
		return high;
	}

	public void setHigh(double high) {
		this.high = high;
	}

	public double getLow() {
		return low;
	}

	public void setLow(double low) {
		this.low = low;
	}

	public double getClose() {
		return close;
	}

	public void setClose(double close) {
		this.close = close;
	}

	public double getVolume() {
		return volume;
	}

	public void setVolume(double volume) {
		this.volume = Math.abs(volume);
		this.profit=this.close-this.open;
	}

	public double getProfit() {
		return profit;
	}

	public void setProfit(double profit) {
		this.profit = profit;
	}
	
	@Override
	public String toString() {
		return "PriceData [open=" + open + ", high=" + high + ", low=" + low + ", close=" + close + ", volume=" + volume
				+ "]";
	}

}

File: PySpark_Crypto/junk/PriceData.py
class PriceData:

    def __init__(self,op=0,hp=0,lp=0,cp=0,vol=0):
        self.__open = float(op)
        self.__high = float(hp)
        self.__low = float(lp)
        self.__close = float(cp)
        self.__volume = abs(float(vol))
    
    def getOpen(self):
        return self.__open    
    def getClose(self):
        return self.__close    
    def getHigh(self):
        return self.__high    
    def getLow(self):
        return self.__low
    def getVolume(self):
        return self.__volume

    def setOpen(self,value):
        self.__open = float(value)    
    def setHigh(self,value):
        self.__high = float(value)        
    def setLow(self,value):
        self.__low = float(value)        
    def setClose(self,value):
        self.__close = float(value)
    def setVolume(self,value):
        self.__volume = abs(float(value))


    def __str__(self):
        return "Price[Open:"+str(self.__open)+",High:"+str(self.__high)+",Low:"+str(self.__low)+",Close:"+str(self.__close)+",Volume:"+str(self.__volume)+"]"
